### PR TITLE
Metric: Discard Error Metric When Taking Snapthot

### PR DESCRIPTION
**Pull request by Pengzna**
as title

### CODE REVIEW

File: iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/reporter/prometheus/PrometheusReporter.java [hunk 1]
```diff
@@ -144,6 +145,10 @@ private String scrape() {
         } else if (metric instanceof Timer) {
           Timer timer = (Timer) metric;
           HistogramSnapshot snapshot = timer.takeSnapshot();
+          if (Objects.isNull(snapshot)) {
+            LOGGER.warn("Detected an error when taking snapshot, will discard this metric");
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/reporter/prometheus/PrometheusReporter.java:149]:
```suggestion
            LOGGER.warn("Detected an error when taking metric timer snapshot, will discard this metric");
```

File: iotdb-core/metrics/core/src/main/java/org/apache/iotdb/metrics/core/type/IoTDBTimer.java [hunk 2]
```diff
@@ -40,37 +45,73 @@ public void update(long duration, TimeUnit unit) {
...
+      LOGGER.warn(
+          "Detected an error while taking snapshot, may cause a miss during this recording.", e);
+      return null;
+    }
   }
 
   @Override
   public double getSum() {
-    return this.takeSnapshot().getSum();
+    HistogramSnapshot snapshot = this.takeSnapshot();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/metrics/core/src/main/java/org/apache/iotdb/metrics/core/type/IoTDBTimer.java:59]:
```
return Optional.ofNullable(takeSnapshot()).map(HistogramSnapshot::getSum).orElse(0.0);
```
use this as you already log in `takeSnapshot`?

File: iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/reporter/prometheus/PrometheusReporter.java (hunk 1 above)
Comment (Pengzna) [reply iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/reporter/prometheus/PrometheusReporter.java:149]: applied

File: iotdb-core/metrics/core/src/main/java/org/apache/iotdb/metrics/core/type/IoTDBTimer.java (hunk 2 above)
Comment (Pengzna) [reply iotdb-core/metrics/core/src/main/java/org/apache/iotdb/metrics/core/type/IoTDBTimer.java:59]: elegant! fixed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**